src/test/java/uk/nhs/hee/tis/trainee/credentials/service/JwtServiceTest.java

```diff
@@ -0,0 +1,164 @@
+    }
+
+    final Instant now = Instant.now();
+    String tokenString = service.generateToken(new EmptyData());
+
+    Jwt<?, Claims> token = parser.parse(tokenString);
+    Claims tokenClaims = token.getBody();
+
+    assertThat("Unexpected number of claims.", tokenClaims.size(), is(DEFAULT_CLAIM_COUNT));
+    assertThat("Unexpected token aud.", tokenClaims.getAudience(), is(AUDIENCE));
+    assertThat("Unexpected token iss.", tokenClaims.getIssuer(), is(ISSUER));
+
+    Date issuedAt = tokenClaims.getIssuedAt();
+    long issuedAtDiff = Duration.between(issuedAt.toInstant(), now).toMinutes();
+    assertThat("Unexpected difference between now and iat.", issuedAtDiff, is(0L));
+    assertThat("Unexpected token nbf.", tokenClaims.getNotBefore(), is(issuedAt));
+
+    Date expectedExpiration = Date.from(issuedAt.toInstant().plus(Duration.ofDays(30)));
+    assertThat("Unexpected token exp.", tokenClaims.getExpiration(), is(expectedExpiration));
+  }
+
+  @Test
+  void shouldGenerateTokenWithProgrammeMembershipClaims() {
+    ProgrammeMembershipDto dto = new ProgrammeMembershipDto(
+        "123", PROGRAMME_NAME, START_DATE, END_DATE);
+
+    String tokenString = service.generateToken(dto);
+
+    Jwt<?, Claims> token = parser.parse(tokenString);
+    Claims tokenClaims = token.getBody();
+
+    assertThat("Unexpected number of claims.", tokenClaims.size(), is(DEFAULT_CLAIM_COUNT + 3));
+    assertThat("Unexpected claim.", tokenClaims.get("tisId"), nullValue());
+    assertThat("Unexpected programme name.", tokenClaims.get("programmeName"), is(PROGRAMME_NAME));
+    assertThat("Unexpected programme start date.", tokenClaims.get("startDate"),
+        is(START_DATE.toString()));
+    assertThat("Unexpected programme end date.", tokenClaims.get("endDate"),
+        is(END_DATE.toString()));
+
+    Instant issuedAt = tokenClaims.getIssuedAt().toInstant();
+    Instant expectedExpiration = dto.getExpiration(issuedAt).truncatedTo(ChronoUnit.SECONDS);
+    Instant expiration = tokenClaims.getExpiration().toInstant();
+    assertThat("Unexpected token exp.", expiration, is(expectedExpiration));
+  }
+
+  @Test
+  void shouldGenerateTokenWithTestCredentialClaims() {
+    TestCredentialDto dto = new TestCredentialDto(GIVEN_NAME, FAMILY_NAME, BIRTH_DATE);
+
+    String tokenString = service.generateToken(dto);
+
+    Jwt<?, Claims> token = parser.parse(tokenString);
+    Claims tokenClaims = token.getBody();
+
+    assertThat("Unexpected number of claims.", tokenClaims.size(), is(DEFAULT_CLAIM_COUNT + 3));
+    assertThat("Unexpected given name.", tokenClaims.get("givenName"), is(GIVEN_NAME));
+    assertThat("Unexpected family name.", tokenClaims.get("familyName"),
+        is(FAMILY_NAME));
+    assertThat("Unexpected birth date.", tokenClaims.get("birthDate"), is(BIRTH_DATE.toString()));
+
+    Instant issuedAt = tokenClaims.getIssuedAt().toInstant();
+    Instant expectedExpiration = dto.getExpiration(issuedAt).truncatedTo(ChronoUnit.SECONDS);
```

Comment:
Interestingly, this works to MILLIS-level truncation as well, but the programmeMembership one doesn't

Reply:
@ReubenRobertsHEE FYI, it's because `LocalTime.MAX` sets the millis and nanos to 9s (`'23:59:59.999999999'`). We lose some precision between that and the integer representation the JWT builder converts to (`'23:59:59.999999000'` iirc) so they end up not matching. The `TestCredential` expiry stays as  an all-zero time, so the loss of precision doesn't affect the comparison.